src/main/java/com/redhat/devtools/intellij/lsp4mp4ij/psi/core/utils/PsiTypeUtils.java

```diff
@@ -178,6 +205,43 @@ public static String getOptionalTypeParameter(String typeName) {
         return typeName.substring(start + 1, end);
     }
 
+    /**
+     * Returns the first (generic) type parameter for the given <code>psiType</code>.
+     * Returns the given <code>psiType</code> if the given psiType has no generic type parameter.
+     *
+     * Examples:
+     * <ul>.
+     *  <li>returns {@code String} for {@code List<String>}</li>
+     *  <li>returns {@code List<String>} for {@code Optional<List<String>>}</li>
+     * </ul>
+     *
+     * @param psiType the psiType to return the generic type parameter for
+     * @return the generic type parameter for the given psiType
+     */
+    public static PsiType getTypeParameter(PsiType psiType) {
```

Comment:
I will rename getTypeParameter to getFirstTypeParameter.

Reply:
I will return null instead of the given PsiType because in your code you check getTypeParameter != null